roles/caw/tasks/main.yml

```diff
@@ -0,0 +1,53 @@
+---
+
+- name: Fetch CAW from GitHub
+  git: repo="{{ caw_repo }}"
+       dest="{{ caw_dest }}"
+       version="{{ caw_version }}"
+       force=yes
+
+- name: Create CAW Singularity folder
+  file: name="{{ caw_singularity }}"
+        state=directory mode=g+s
+
+- name: Create temp directory for CAW singularity images
+  command: "mkdir {{ item }}"
+  args:
+    creates: "{{ item }}"
+  with_items:
+  - "{{ caw_singularity }}/singularity_cache"
+  - "{{ caw_singularity }}/singularity_temp"
+
+- name: Pull CAW Singularity images
+  command: "singularity build {{ item }}-{{ caw_version }}.img {{ caw_singularity_repo }}/{{ item }}:{{ caw_version }}"
+  environment:
+    SINGULARITY_CACHEDIR: "{{ caw_singularity }}/singularity_cache"
+    SINGULARITY_TMPDIR: "{{ caw_singularity }}/singularity_temp"
+  args:
+    chdir: "{{ caw_singularity }}"
+  with_items: "{{ caw_singularity_images }}"
+  ignore_errors: true
+
+- name: Remove CAW singularity temp directories
+  command: "rm -rf {{ item }}" 
```

Comment:
Considering how many shell commands you're using instead of the `file` module I'm guessing I might be missing something here, but otherwise I would have suggested the `state=absent` argument instead here.

Reply:
Again no specific, since I am new to `ansible` I am only using the obvious/simple modules. So `file` with `state=absent` removes the directory ? That is really cool, I will change this as well 👍